frontend2/src/components/compete/ScrimmagingRecord.tsx

```diff
@@ -0,0 +1,106 @@
+  hideUnranked?: boolean;
+  hideRanked?: boolean;
+  className?: string;
+}
+
+const ScrimmagingRecord: React.FC<ScrimmagingRecordProps> = ({
+  team,
+  hideTeamName = false,
+  hideAllScrimmages = false,
+  hideUnranked = false,
+  hideRanked = false,
+  className = "",
+}) => {
+  const { episodeId } = useEpisodeId();
+  const scrimmagingRecordAll = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType: CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.All,
+  });
+  const scrimmagingRecordUnranked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Unranked,
+  });
+  const scrimmagingRecordRanked = useScrimmagingRecord({
+    episodeId,
+    teamId: team.id,
+    scrimmageType:
+      CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Ranked,
+  });
+
+  return (
+    <div
+      className={`flex h-full w-full flex-col items-start gap-1 ${className}`}
+    >
+      {!hideTeamName && (
+        <div className="flex flex-1 flex-row items-center justify-start gap-4">
+          <img
+            className="h-20 w-20 rounded-full bg-gray-400 md:h-16 md:w-16"
+            src={team.profile?.avatar_url}
+          />
+          <div className="text-xl font-semibold">{team.name}</div>
+        </div>
+      )}
+      {scrimmagingRecordAll.isError ||
+      scrimmagingRecordUnranked.isError ||
+      scrimmagingRecordRanked.isError ? (
+        <div className="mb-2 mt-4 flex w-full flex-row items-center justify-center gap-3 text-gray-400">
+          Error fetching scrimmaging record.
+        </div>
+      ) : (
+        <div className="flex w-full flex-row items-center justify-center gap-3 md:flex-col">
+          {!hideAllScrimmages && (
+            <WinLossTie
+              scrimmageType={
+                CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.All
+              }
+              loading={scrimmagingRecordAll.isFetching}
+              wins={scrimmagingRecordAll.data?.wins ?? 0}
+              losses={scrimmagingRecordAll.data?.losses ?? 0}
+              ties={scrimmagingRecordAll.data?.ties ?? 0}
+            />
+          )}
+          {!hideUnranked && (
+            <WinLossTie
+              scrimmageType={
+                CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Unranked
+              }
+              loading={scrimmagingRecordUnranked.isFetching}
+              wins={scrimmagingRecordUnranked.data?.wins ?? 0}
+              losses={scrimmagingRecordUnranked.data?.losses ?? 0}
+              ties={scrimmagingRecordUnranked.data?.ties ?? 0}
+            />
+          )}
+          {!hideRanked && (
+            <WinLossTie
+              scrimmageType={
+                CompeteMatchScrimmagingRecordRetrieveScrimmageTypeEnum.Ranked
+              }
+              loading={scrimmagingRecordRanked.isFetching}
+              wins={scrimmagingRecordRanked.data?.wins ?? 0}
```

Comment:
We can reduce the repetitiveness of this code by moving the useScrimmagingRecord call into the WinLossTie component

Reply:
Done!